pkg/node/events.go

```diff
@@ -130,6 +130,28 @@ func ListenEvents(
 		}
 	}
 
-	return nil,
-		fmt.Errorf("listening events for: opId %s, caller %s, emitter %s: Timeout", *operationID, *caller, *emitter)
+	// Handle the case where one or more pointers are nil
+	return listenEventError(operationID, caller, emitter)
+}
+
+func listenEventError(operationID *string, caller *string, emitter *string) ([]Event, error) {
+	if operationID == nil || caller == nil || emitter == nil {
+		err := "timeout while listening events for: "
+		if operationID == nil {
+			err += "operationID is nil, "
+		}
+
+		if caller == nil {
+			err += "caller is nil, "
+		}
+
+		if emitter == nil {
+			err += "emitter is nil, "
+		}
+
+		return nil, fmt.Errorf(err)
+	}
+
+	return nil, fmt.Errorf(
+		"timeout while listening events for: opId %s, caller %s, emitter %s", *operationID, *caller, *emitter)
```

Comment:
not very conviced by this function. maybe its normal that these pointers are nil in case of error. in that case just remove them from the error message

Reply:
these pointers can be nil, yes, the goal here is to prevent panic while formatting the error.